expr/func_aspercent.go

```diff
@@ -158,6 +159,16 @@ func (s *FuncAsPercent) execWithNodes(in, totals []models.Series, dataMap DataMa
 				// key found in both inByKey and totalSerieByKey
 				serie1, serie2 := NormalizeTwo(serie1, totalSerieByKey[key], NewCOWCycler(dataMap))
 				serie1 = serie1.Copy(pointSlicePool.Get())
+
+				// this should not happen
+				if len(serie1.Datapoints) != len(serie2.Datapoints) {
+					log.Errorf("AsPercent: len of serie1 datapoints (%v) does not match len of serie2 datapoints (%v) - truncating", len(serie1.Datapoints), len(serie2.Datapoints))
+					if len(serie1.Datapoints) > len(serie2.Datapoints) {
+						serie1.Datapoints = serie1.Datapoints[:len(serie2.Datapoints)]
+					}
```

Comment:
what if one of them has the extra point at the beginning? this would cause all of them to be mis-aligned.

Reply:
True

Reply:
after further review, based on how the Normalize* functions work and how everything is processed, I am confident that the extra point will always be at the end. I would like to move forward with this as is.